tests/system/libraries/SystemTestSpy/windows.py

```diff
@@ -32,50 +52,89 @@ def _GetWindows(
 		filterUsingWindow: Callable[[Window], bool] = lambda _: True,
 ) -> List[Window]:
 	windows: List[Window] = []
-	EnumWindowsProc = WINFUNCTYPE(c_bool, POINTER(c_int), POINTER(c_int))
 
-	def _append_title(hwnd: HWND, _lParam: LPARAM) -> bool:
+	# BOOL CALLBACK EnumWindowsProc _In_ HWND,_In_ LPARAM
+	# HWND as a pointer creates confusion, treat as an int
+	# http://makble.com/the-story-of-lpclong
+	@WINFUNCTYPE(c_bool, c_int, POINTER(c_int))
+	def _append_title(hwnd: HWNDVal, _lParam: LPARAM) -> bool:
+		if not isinstance(hwnd, (HWNDVal, c_int)):
+			builtIn.log(f"Hwnd type {type(hwnd)}, value {hwnd}")
 		window = Window(hwnd, _GetWindowTitle(hwnd))
 		if filterUsingWindow(window):
 			windows.append(window)
 		return True
 
-	if not windll.user32.EnumWindows(EnumWindowsProc(_append_title), 0):
+	if not windll.user32.EnumWindows(_append_title, 0):
 		raise WinError()
 	return windows
 
 
 def _GetVisibleWindows() -> List[Window]:
 	return _GetWindows(
-		filterUsingWindow=lambda window: windll.user32.IsWindowVisible(window.hwnd) and bool(window.title)
+		filterUsingWindow=lambda window: windll.user32.IsWindowVisible(window.hwndVal) and bool(window.title)
 	)
 
 
-def SetForegroundWindow(targetTitle: re.Pattern, logger: Callable[[str], None] = lambda _: None) -> bool:
-	currentTitle = GetForegroundWindowTitle()
-	if re.match(targetTitle, currentTitle):
-		logger(f"Window '{currentTitle}' already focused")
+def CloseWindow(window: Window) -> bool:
+	"""
+	@return: True if the window exists and the message was sent.
+	"""
+	if windowWithHandleExists(window.hwndVal):
+		return windll.user32.CloseWindow(
+			window.hwndVal,
+		)
+	return False
+
+
+Logger = Callable[[str], None]
+
+
+def SetForegroundWindow(window: Window, logger: Logger) -> bool:
+	"""This may be unreliable, there are conditions on which processes can set the foreground window.
+	https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow#remarks
+	Additionally, note that this code is run by the Robot Framework test runner, not NVDA.
+	"""
+	if window is None or Window.hwndVal is None:
```

Comment:
I assume this should be the instance variable. Additionally, should `Window.hwndVal` be retyped as `Optional[HWNDVal]`?

```suggestion
	if window is None or window.hwndVal is None:
```